src/main/java/org/graylog2/gelfclient/transport/GelfSenderThread.java

```diff
@@ -139,4 +146,40 @@ public void stop() {
         keepRunning.set(false);
         senderThread.interrupt();
     }
+
+    /**
+     * Block and wait for all messages in the queue to send. This can be used during shutdown to wait for
+     * queued messages to send before shutting down.
+     * Waiting will continue for the indicated {@code FLUSH_WAIT_RETRIES} and {@code FLUSH_WAIT_MS.}
+     */
+    void flushSynchronously() {
+
+        for (int i = 0; i <= FLUSH_WAIT_RETRIES; i++) {
+            if (!flushingInProgress()) {
+                LOG.debug("Successfully flushed messages. Shutting down now.");
+                continue;
+            }
+
+            LOG.debug("Flushing in progress. [{}] messages are still enqueued, and [{}] messages are still in-flight.",
+                      queue.size(), inflightSends.get());
+
+            try {
+                Thread.sleep(FLUSH_WAIT_MS);
+            } catch (InterruptedException e) {
+                LOG.error("Interrupted message flushing during shutdown after [{}}] attempts.", i);
```

Comment:
I think we should `break` here instead of executing the next iteration of the `for` loop. When the sleep is interrupted, something tells us to stop. :smile: 

We should probably also re-interrupt the thread with `Thread.currentThread().interrupt()` instead of just swallowing the exception. (See ["Don't swallow interrupts"](https://www.ibm.com/developerworks/library/j-jtp05236/index.html))